pkg/odo/cli/component/create.go

```diff
@@ -881,6 +881,101 @@ func (co *CreateOptions) Validate() (err error) {
 	return nil
 }
 
+func downloadGitProject(starterProject *devfilev1.StarterProject, starterToken, path string) error {
+
+	var projectSource devfilev1.GitLikeProjectSource
+	if starterProject.Git != nil {
+		projectSource = starterProject.Git.GitLikeProjectSource
+	} else {
+		projectSource = starterProject.Github.GitLikeProjectSource
+	}
+
+	remoteName, remoteUrl, revision, err := parsercommon.GetDefaultSource(projectSource)
+	if err != nil {
+		return errors.Wrapf(err, "unable to get default project source for starter project %s", starterProject.Name)
+	}
+
+	refName := plumbing.ReferenceName(revision)
+
+	if plumbing.IsHash(revision) {
+		log.Warning("Specifying commit in 'revision' is not yet supported in odo.")
+		// overriding revision to empty as we do not support this
+		revision = ""
+	}
+
+	if revision != "" {
+		// lets consider revision to be a branch name first
+		refName = plumbing.NewBranchReferenceName(revision)
+	}
+
+	downloadSpinner := log.Spinnerf("Downloading starter project %s from %s", starterProject.Name, remoteUrl)
```

Comment:
Using `%q` would print better output. I know we don't follow this across our code. So leaving it up to you to decide if you want to do it or not.